pkg/auth/signatures.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"io"
+	"os"
+
+	"github.com/square/p2/pkg/util"
+	"golang.org/x/crypto/openpgp"
+)
+
+// Something that can verify PGP signatures
+type SigVerifier interface {
+	CheckDetachedSignature(signed, signature io.Reader) (*openpgp.Entity, error)
+}
+
+// A SignatureVerifier that reads its keyring from a file. The file is
+// read every time it is needed.
```

Comment:
The way to do that is fairly straightforward, but I'm not convinced yet that it's really required.  How frequently does a signature need to be verified?  How big is this keyring going to get?  I suspect that it isn't worth the work to make this more sophisticated.